sympy/functions/elementary/piecewise.py

```diff
@@ -774,6 +774,9 @@ def nonsymfail(cond):
                 if rhs is not None and cond not in (S.true, S.false):
                     lower = upper = rhs
                     cond = cond.subs(sym, lower)
+                    if cond is S.true:
+                        cond = Eq(sym, lower)
```

Comment:
Very nice tweak.

Instead of making cond an Eq, do with it what is done above:

```
if cond is S.true:
    int_expr.append((rhs, rhs, expr.subs(sym, rhs), iarg))
    continue
```

This will still lead to a duplicate but I'm not sure this routine has to worry about that. At worst, perhaps, `uniq` could be applied at the end before returning the result because I don't know if it would be possible for duplicates to arise in other ways:

```
return list(uniq(int_expr))
```

Reply:
@smichr I have changed both things. I didn't saw any test failing with unique elements, so I don't think that there is any need for redundant tuples.